sktime/markov/msm/augmented_msm.py

```diff
@@ -0,0 +1,623 @@
+
+        Returns
+        -------
+        estimator : AugmentedMSMEstimator
+            An estimator parameterized expectations by state based on feature trajectories.
+        """
+        discrete_trajectories = ensure_dtraj_list(discrete_trajectories)
+        feature_trajectories = ensure_traj_list(feature_trajectories, np.float32)
+        # check input
+        if np.all(sigmas > 0):
+            _w = 1. / (2 * sigmas ** 2.)
+        else:
+            raise ValueError('Zero or negative standard errors supplied. Please revise input')
+        if feature_trajectories[0].ndim < 2:
+            raise ValueError("Supplied feature trajectories have inappropriate dimensions (%d), "
+                             "should be at least 2." % feature_trajectories[0].ndim)
+        if len(discrete_trajectories) != len(feature_trajectories):
+            raise ValueError("A different number of dtrajs and ftrajs were supplied as input. "
+                             "They must have exactly a one-to-one correspondence.")
+        if not np.all([len(dt) == len(ft) for dt, ft in zip(discrete_trajectories, feature_trajectories)]):
+            raise ValueError("One or more supplied dtraj-ftraj pairs do not have the same length.")
+        # MAKE E matrix
+        fta = np.concatenate(feature_trajectories)
+        dta = np.concatenate(discrete_trajectories)
+        _E = np.zeros((n_states, fta.shape[1]))
+        for i, s in enumerate(range(n_states)):
+            indices = np.where(dta == s)
+            if len(indices[0]) > 0:
+                _E[i, :] = fta[indices].mean(axis=0)
+        # transition matrix estimator
+        return AugmentedMSMEstimator(expectations_by_state=_E, experimental_measurements=experimental_measurements,
+                                     experimental_measurement_weights=_w, eps=eps,
+                                     support_ci=support_ci, maxiter=maxiter, max_cache=max_cache)
+
+    @property
+    def expectations_by_state(self):
+        r""" The expectations by state (N) for each observable (K).
+
+        :type: (N, K) ndarray
+        """
+        return self._E
+
+    @expectations_by_state.setter
+    def expectations_by_state(self, value):
+        self._E = value
+
+    @property
+    def experimental_measurements(self):
+        r""" Experimental measurement averages (K).
+
+        :type: (K,) ndarray
+        """
+        return self._m
+
+    @experimental_measurements.setter
+    def experimental_measurements(self, value):
+        self._m = value
+
+    @property
+    def experimental_measurement_weights(self):
+        r""" Weights for experimental measurement averages (K).
+
+        :type: (K,) ndarray
+        """
+        return self._w
+
+    @experimental_measurement_weights.setter
+    def experimental_measurement_weights(self, value):
+        if np.any(value < 1e-12):
+            raise ValueError("Some weights are close to zero or negative, but only weights greater or equal 1e-12 can"
```

Comment:
curious why `1e-12` specifically as opposed to a tunable tolerance (but doesn't matter)

Reply:
This is quite arbitrary indeed, took it from the original code. Perhaps an empirical value, we could ask simon

Reply:
no need to let this hold up the merge, but it seems that more general would be better

Reply:
same for the below ones too i'll just keep the convo here